repair/repair.cc

```diff
@@ -1311,6 +1319,13 @@ future<> repair_service::sync_data_using_repair(
     co_await task->done();
 }
 
+double data_sync_repair_task_impl::get_total() const noexcept {
+    auto module = dynamic_pointer_cast<repair_module>(_module);
+    auto& db = module->get_repair_service().get_db().local();
+    auto cfs = list_column_families(db, _status.keyspace);
+    return _ranges.size() * cfs.size() * smp::count;
```

Comment:
In data_sync_repair_task_impl::run, we have the logic to list the tables we need to repair for the keyspace. Can we remove the duplication in data_sync_repair_task_impl::get_total() ?

Reply:
The main reason why I did it like this is that before this line is reached:
https://github.com/scylladb/scylladb/blob/d7f56d736764a37c36d37c2b882aef730f014d59/repair/repair.cc#L1340
user can already request task status and `cfs.size()` is needed for calculating total progress. 

If the main problem here is time consumed by `list_column_families()` then it can be done before the task is created and passed to `data_sync_repair_task_impl` constructor. I am not sure if this is ok, since list of cf sounds like something pretty dynamic, is it? Oh, if it is, then current get_total would return different result each time its called...

If it's about the whole:
```
auto module = dynamic_pointer_cast<repair_module>(_module);
auto& db = module->get_repair_service().get_db().local();
```
Then we could set the `cfs.size()` with setter and return 0.0/1.0 progress if it hasn't been set yet. This would slightly miss the "progress_units="ranges" but I think it's not that bad. Would that be ok?

Reply:
Yes, I know that the number of tables to be repaired is only called inside data_sync_repair_task_impl::run. It is important to get a consistent number of tables in get_total() as the task will will repairs in data_sync_repair_task_impl::run.

We can return 0% if  we haven't reached the run() where the correct get_total is calculated. It is ok to "miss" some progress since user won't be able to query all the progresses anyway. 

We can set the total size in run(), so get_total() will be faster too.